config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -170,13 +165,9 @@ public void getAllConfigs(
 
       responseObserver.onNext(responseBuilder.build());
       responseObserver.onCompleted();
-    } catch (StatusRuntimeException e) {
-      log.error("Get all configs failed for request: {}", request, e);
-      responseObserver.onError(e);
     } catch (Exception e) {
       log.error("Get all configs failed for request: {}", request, e);
-      responseObserver.onError(
-          Status.INTERNAL.withCause(e).withDescription("Unexpected error").asRuntimeException());
+      responseObserver.onError(Status.fromThrowable(e).withCause(e).asRuntimeException());
```

Comment:
fromThrowable already sets the cause if necessary

```suggestion
      responseObserver.onError(Status.fromThrowable(e).asRuntimeException());
```

Reply:
fixed